output/changelog.go

```diff
@@ -0,0 +1,315 @@
+		return nil, summaryGenerationErrCode, fmt.Errorf("failed to print summary in %s format with %w", format, err)
+	}
+	return diffOutput, noErrCode, nil
+}
+
+func (p *changelogProfile) checkAddedPaths() checker.BackwardCompatibilityErrors {
+	errs := checker.BackwardCompatibilityErrors{}
+	for _, path := range p.diffReport.PathsDiff.Added {
+		dr := &diff.Diff{
+			PathsDiff: &diff.PathsDiff{
+				Added:    []string{path},
+				Base:     p.diffReport.PathsDiff.Base,
+				Revision: p.diffReport.PathsDiff.Revision,
+			},
+		}
+		errs = append(errs, p.check(dr)...)
+	}
+	return errs
+}
+
+func (p *changelogProfile) checkDeletedPaths() checker.BackwardCompatibilityErrors {
+	errs := checker.BackwardCompatibilityErrors{}
+	for _, path := range p.diffReport.PathsDiff.Deleted {
+		dr := &diff.Diff{
+			PathsDiff: &diff.PathsDiff{
+				Deleted:  []string{path},
+				Base:     p.diffReport.PathsDiff.Base,
+				Revision: p.diffReport.PathsDiff.Revision,
+			},
+		}
+		errs = append(errs, p.check(dr)...)
+	}
+	return errs
+}
+
+func (p *changelogProfile) checkModifiedPaths() checker.BackwardCompatibilityErrors {
+	errs := checker.BackwardCompatibilityErrors{}
+	for path, pathDiff := range p.diffReport.PathsDiff.Modified {
+		errs = append(errs, p.checkModifiedRefDiff(path, pathDiff)...)
+		errs = append(errs, p.checkModifiedSummaryDiff(path, pathDiff)...)
+		errs = append(errs, p.checkModifiedDescriptionDiff(path, pathDiff)...)
+		errs = append(errs, p.checkModifiedOperationsDiff(path, pathDiff)...)
+
+		// todo: pathDiff.ExtensionsDiff
+		// todo: pathDiff.ServersDiff
+		// todo: pathDiff.ParametersDiff
+
+		// dr := &diff.Diff{
+		// 	PathsDiff: &diff.PathsDiff{
+		// 		Modified: map[string]*diff.PathDiff{path: pathDiff},
+		// 		Base:     p.diffReport.PathsDiff.Base,
+		// 		Revision: p.diffReport.PathsDiff.Revision,
+		// 	},
+		// }
+
+		// errs = append(errs, p.check(dr)...)
+	}
+	return errs
+}
+
+func (p *changelogProfile) checkModifiedRefDiff(path string, pathDiff *diff.PathDiff) checker.BackwardCompatibilityErrors {
+	if pathDiff.RefDiff == nil {
+		return checker.BackwardCompatibilityErrors{}
+	}
+	return p.checkModifiedPathDiff(path, &diff.PathDiff{RefDiff: pathDiff.RefDiff})
+}
+
+func (p *changelogProfile) checkModifiedSummaryDiff(path string, pathDiff *diff.PathDiff) checker.BackwardCompatibilityErrors {
+	if pathDiff.SummaryDiff == nil {
+		return checker.BackwardCompatibilityErrors{}
+	}
+	return p.checkModifiedPathDiff(path, &diff.PathDiff{SummaryDiff: pathDiff.SummaryDiff})
+}
+
+func (p *changelogProfile) checkModifiedDescriptionDiff(path string, pathDiff *diff.PathDiff) checker.BackwardCompatibilityErrors {
+	if pathDiff.DescriptionDiff == nil {
+		return checker.BackwardCompatibilityErrors{}
+	}
+	return p.checkModifiedPathDiff(path, &diff.PathDiff{DescriptionDiff: pathDiff.DescriptionDiff})
+}
+
+func (p *changelogProfile) checkModifiedOperationsDiff(path string, pathDiff *diff.PathDiff) checker.BackwardCompatibilityErrors {
```

Comment:
the most complex part, this code is wip, just to evaluate the degree of complexity

Reply:
Should be similar to the other checkers